execution_engine/src/engine_state/mod.rs

```diff
@@ -2192,20 +2214,20 @@ where
     }
 
     /// Gets a trie object for given state root hash.
-    pub fn get_trie_full(&self, trie_key: Digest) -> Result<Option<TrieRaw>, Error>
-    where
-        Error: From<S::Error>,
-    {
-        Ok(self.state.get_trie_full(&trie_key)?)
+    pub fn get_trie_full(&self, trie_key: Digest) -> Result<Option<TrieRaw>, Error> {
+        match self.state.get_trie_full(&trie_key) {
+            Ok(ret) => Ok(ret),
+            Err(err) => Err(err.into()),
+        }
     }
 
     /// Puts a trie if no children are missing from the global state; otherwise reports the missing
     /// children hashes via the `Error` enum.
-    pub fn put_trie_if_all_children_present(&self, trie_bytes: &[u8]) -> Result<Digest, Error>
-    where
-        Error: From<S::Error>,
-    {
-        let missing_children = self.state.missing_children(trie_bytes)?;
+    pub fn put_trie_if_all_children_present(&self, trie_bytes: &[u8]) -> Result<Digest, Error> {
+        let missing_children = match self.state.missing_children(trie_bytes) {
```

Comment:
Nit: Same here, could we use the `?` operator

Reply:
this function will be deleted in an upcoming PR anyway.